### PR TITLE
Added flink k8s operator validate

### PR DESCRIPTION
**Pull request by ChengJie1053**
Added flink k8s operator validate,Check whether FlinkDeployment CRD is deployed in the k8s cluster

### CODE REVIEW

**Comment by ChengJie1053** [comment]
@Al-assad

**Comment by ChengJie1053** [comment]
This pr is used to verify that the flink operator crd is deployed on the k8s cluster

File: streampark-flink/streampark-flink-kubernetes-v2/streampark-flink-kubernetes-core/src/main/scala/org/apache/streampark/flink/kubernetes/v2/operator/CROperator.scala [hunk 1]
```diff
@@ -58,6 +60,7 @@ object CROperator extends CROperator {
   /** Apply FlinkDeployment CR. */
   // noinspection DuplicatedCode
   override def applyDeployment(spec: FlinkDeploymentDef): IO[Throwable, Unit] = {
+    validateDeployFlinkDeploymentCRD
```
Comment (Al-assad) [inline streampark-flink/streampark-flink-kubernetes-v2/streampark-flink-kubernetes-core/src/main/scala/org/apache/streampark/flink/kubernetes/v2/operator/CROperator.scala]: `validateDeployFlinkDeploymentCRD` return an IO structure that will not be executed here.

**Review by Al-assad — CHANGES_REQUESTED** [review]

File: streampark-flink/streampark-flink-kubernetes-v2/streampark-flink-kubernetes-core/src/main/scala/org/apache/streampark/flink/kubernetes/v2/operator/CROperator.scala (hunk 1 above)
Comment (ChengJie1053) [reply streampark-flink/streampark-flink-kubernetes-v2/streampark-flink-kubernetes-core/src/main/scala/org/apache/streampark/flink/kubernetes/v2/operator/CROperator.scala]: Ok, thanks for reviewing the code

File: streampark-flink/streampark-flink-kubernetes-v2/streampark-flink-kubernetes-core/src/main/scala/org/apache/streampark/flink/kubernetes/v2/operator/CROperator.scala [hunk 2]
```diff
@@ -58,6 +60,10 @@ object CROperator extends CROperator {
   /** Apply FlinkDeployment CR. */
   // noinspection DuplicatedCode
   override def applyDeployment(spec: FlinkDeploymentDef): IO[Throwable, Unit] = {
+    for {
+      _ <- validateDeployFlinkDeploymentCRD
```
Comment (Al-assad) [inline streampark-flink/streampark-flink-kubernetes-v2/streampark-flink-kubernetes-core/src/main/scala/org/apache/streampark/flink/kubernetes/v2/operator/CROperator.scala]: The effect of this code segment will actually be ignored, it should move to line 69

**Review by Al-assad — CHANGES_REQUESTED** [review]
validateDeployFlinkDeploymentCRD

File: streampark-flink/streampark-flink-kubernetes-v2/streampark-flink-kubernetes-core/src/main/scala/org/apache/streampark/flink/kubernetes/v2/operator/CROperator.scala (hunk 2 above)
Comment (ChengJie1053) [reply streampark-flink/streampark-flink-kubernetes-v2/streampark-flink-kubernetes-core/src/main/scala/org/apache/streampark/flink/kubernetes/v2/operator/CROperator.scala]: Ok, thanks for reviewing the code

**Review by Al-assad — APPROVED** [review]
LGTM

**Pull request merged**